src/transform/ir_proto_to_datalog/arcs_manifest_tree/recipe.cc

```diff
@@ -0,0 +1,75 @@
+#include "src/transform/ir_proto_to_datalog/arcs_manifest_tree/recipe.h"
+#include "src/transform/ir_proto_to_datalog/logging/context_logger.h"
+
+#include "third_party/arcs/proto/manifest.pb.h"
+
+#include "absl/strings/string_view.h"
+
+#include <memory>
+#include <string>
+#include <cstdint>
+
+namespace raksha::transform::arcs_manifest_tree {
+
+namespace logging = ::raksha::transform::logging;
+
+Recipe::Recipe(const absl::string_view name, const uint64_t num_handles)
```

Comment:
I don't think we should be replicating the arcs manifest as is in C++. We want to introduce a level of abstraction like graph (which is that the recipe is).